src/components/Amalgamation/BusinessTable.vue

```diff
@@ -69,91 +69,49 @@
 </template>
 
 <script lang="ts">
-import { Component, Emit, Vue, Watch } from 'vue-property-decorator'
-import { Getter, Action } from 'pinia-class'
+import { Component, Emit, Mixins, Watch } from 'vue-property-decorator'
+import { Action } from 'pinia-class'
 import { getName } from 'country-list'
 import { useStore } from '@/store/store'
 import { AmalgamatingStatuses, AmlRoles } from '@/enums'
 import { AmalgamatingBusinessIF } from '@/interfaces'
 import { BaseAddress } from '@bcrs-shared-components/base-address'
 import BusinessStatus from './BusinessStatus.vue'
-import { CorpTypeCd, GetCorpFullDescription } from '@bcrs-shared-components/corp-type-module'
+import { GetCorpFullDescription } from '@bcrs-shared-components/corp-type-module'
+import { AmalgamationMixin } from '@/mixins'
 
 @Component({
   components: {
     BaseAddress,
     BusinessStatus
   }
 })
-export default class BusinessTable extends Vue {
+export default class BusinessTable extends Mixins(AmalgamationMixin) {
   readonly AmlRoles = AmlRoles
   readonly GetCorpFullDescription = GetCorpFullDescription
 
-  @Getter(useStore) getAmalgamatingBusinesses!: AmalgamatingBusinessIF[]
-  @Getter(useStore) isRoleStaff!: boolean
-  @Getter(useStore) isTypeBcCcc!: boolean
-  @Getter(useStore) isTypeBcUlcCompany!: boolean
-
   @Action(useStore) setAmalgamatingBusinesses!: (x: AmalgamatingBusinessIF[]) => void
   @Action(useStore) setDefineCompanyStepValidity!: (x: boolean) => void
 
-  // *** I'M STILL WONDERING IF I WANT TO USE THESE
-  // readonly isLear = (item: AmalgamatingBusinessIF): boolean => (item?.type === 'lear')
-  // readonly isForeign = (item: AmalgamatingBusinessIF): boolean => (item?.type === 'foreign')
-
-  /** True if there a limited company in the table. */
-  get isAnyLimited (): boolean {
-    return this.businesses.some(business =>
-      (business.type === 'lear' && business.legalType === CorpTypeCd.BC_COMPANY)
-    )
-  }
-
   /**
    * This is the list of amalgamating businesses with computed statuses.
-   * In other words, these are the business rules.
+   * In other words, this is where the business rules are evaluated.
    */
   get businesses (): AmalgamatingBusinessIF[] {
```

Comment:
Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce

Reply:
Love this.

Reply:
One architecture/design thing I wanted to mention:

This is a computed value (local getter), which is reactive to the variables used in it, in particular `getAmalgamatingBusinesses`, so it is evaluated (and cached) accordingly.

Also, as a computed, it needs to be used somewhere (otherwise it is not evaluated). In this component, it's used both to display the table items (line 17) and is watched to emit whether the table is valid (line 166).

Another way to do this would be to simply watch `getAmalgamatingBusinesses` and then compute the statuses when it has changed. If the getter approach didn't work (eg, reactivity didn't work) then I would have changed to this approach, otherwise I see no reason to change working code :)